### PR TITLE
[Grid] Add dispatching events in grid definition converter

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | no |
| BC breaks?      | no |
| Related tickets |  |
| License         | MIT |

### CODE REVIEW

**Comment by pjedrzejewski** [comment]
I am not sure it should be in the converter. I had a quick look and there is no obvious place, but we should add it to the place where grid definition is taken from the provider. Otherwise it works only for ArrayGridProvider. WDYT?

File: docs/customization/grid.rst [hunk 1]
```diff
@@ -127,6 +127,67 @@ See an example of fields order modification on the `sylius_admin_product_review`
                     author:
                         position: 4
 
+Customize grids by events
+-------------------------
+
+There is also a second way to customize grids: **via events**.
```
Comment (TheMadeleine) [inline docs/customization/grid.rst]: `There is also another way`

File: docs/customization/grid.rst [hunk 2]
```diff
@@ -127,6 +127,67 @@ See an example of fields order modification on the `sylius_admin_product_review`
...
+Every grid configuration dispatches an event when its definiton is converting.
```
Comment (TheMadeleine) [inline docs/customization/grid.rst]: `when its definition is being converted.`

File: docs/customization/grid.rst [hunk 3]
```diff
@@ -127,6 +127,67 @@ See an example of fields order modification on the `sylius_admin_product_review`
...
+
+For example, **sylius_admin_product** grid dispatches an event:
```
Comment (TheMadeleine) [inline docs/customization/grid.rst]: `dispatches such an event:`

File: docs/customization/grid.rst [hunk 4]
```diff
@@ -127,6 +127,67 @@ See an example of fields order modification on the `sylius_admin_product_review`
...
+
+.. code-block:: php
+
+    sylius.grid.admin_product # For the grid of products in admin
+
+**1.** In order to remove fields from the product grid in **Sylius** you have to create a ``AppBundle\Grid\AdminProductsGridListener`` class.
```
Comment (TheMadeleine) [inline docs/customization/grid.rst]: Before presenting the tutorial we should write what are we going to do:

`To show you and example of a grid customization using events, 
we will remove a field from a grid using that method. Here are the steps, that you need to take:`

File: docs/customization/grid.rst [hunk 5]
```diff
@@ -127,6 +127,67 @@ See an example of fields order modification on the `sylius_admin_product_review`
...
+
+In the example below we are removing an **images** field from ``sylius_admin_product`` grid.
```
Comment (TheMadeleine) [inline docs/customization/grid.rst]: ```the ``images`` field from the ``sylius_admin_product`` grid. ```

File: docs/customization/grid.rst [hunk 6]
```diff
@@ -127,6 +127,67 @@ See an example of fields order modification on the `sylius_admin_product_review`
...
+
+.. code-block:: yaml
+
+    # app/config/config.yml
+    imports:
+        - { resource: "services.yml" }
+
+**3.** Result:
+
+After these two steps your admin product grid should not have image field.
```
Comment (TheMadeleine) [inline docs/customization/grid.rst]: ``` the ``image`` field. ```

**Comment by pjedrzejewski** [comment]
Thanks Grzesiu!

**Pull request merged**